bin/epaa.py

```diff
@@ -591,6 +609,297 @@ def write_empty_files(args: argparse.Namespace):
+    actual combined mutation --> "p.Glu81Leu"
+    Without phasing, VEP does not annotate these together, so we have to do it manually for now
+
+    Args:
+        variant_details_protein: A list of protein variant details.
+        positions: A list of positions corresponding to the mutations in the protein sequence.
+        consequences: A list of consequences for the mutations.
+        seq_wt: The wild-type protein sequence.
+        seq_mut: The mutant protein sequence.
+
+    Returns:
+        List[str]: The updated variant details.
+        List[str]: The updated consequences if provided.
+    """
+    # Input validation
+    if not variant_details_protein or not positions:
+        return variant_details_protein
+
+    if len(variant_details_protein) != len(positions):
+        raise ValueError("Length of variant_details_protein must match length of positions")
+
+    # Prepare lookup for single-letter to three-letter amino acid codes
+    aa1to3 = IUPACData.protein_letters_1to3.copy()
+    aa1to3["*"] = "Ter"
+
+    # Group positions by their list index values to find duplicates
+    position_groups = {}
+    for i, pos in enumerate(positions):
+        if pos not in position_groups:
+            position_groups[pos] = []
+        position_groups[pos].append(i)
+
+    # Collect updated variants
+    updated_variants = []
+    # And, if consequences are provided, update them accordingly
+    if consequences is not None and len(consequences) == len(variant_details_protein):
+        updated_consequences = []
+    else:
+        updated_consequences = None
+    # And, for sorting, updated positions
+    updated_positions = []
+
+    def update_with_original(indices):
+        """
+        Helper function to update variants, positions and consequences for a given set of indices with the original input.
+        """
+        for i in indices:
+            updated_variants.append(variant_details_protein[i])
+            updated_positions.append(positions[i])
+            if updated_consequences is not None:
+                updated_consequences.append(consequences[i])
+
+    # Iterate over the grouped positions
+    for pos, indices in position_groups.items():
+        if len(indices) == 1:
+            # Single variant at this position, we can simply take it
+            update_with_original(indices)
+        else:
+            # Multiple variants at the same position - need to combine
+            try:
+                # Ensure position is within wildtype sequence bounds
+                if pos < 1 or pos > len(seq_wt):
+                    # Keep original variants if position is out of bounds
+                    update_with_original(indices)
+                    continue
+
+                # Ensure variants are missense, stop gained, or synonymous
+                if consequences is not None:
+                    if any([consequences[i] not in ["missense_variant", "stop_gained", "synonymous_variant"] for i in indices]):
```

Comment:
Just to double-check: do we want to exclude other types such as dels/ins, frameshifts here?